src/nix/store-info.cc

```diff
@@ -41,9 +41,9 @@ struct CmdPingStore : StoreCommand, MixJSON
             if (auto version = store->getVersion())
                 res["version"] = *version;
             if (auto trusted = store->isTrustedClient())
-                res["trusted"] = *trusted;
+                res["trusted"] = *trusted ? true : false;
```

Comment:
We should eventually not need this via the `enum struct` change I proposed for `TrustedFlag`, but it is fine to do this for now and that in a follow-up PR.

Reply:
sure.  I saw that TrustedFag is storing only bool value.  But, somehow the value is printing 1/0 instead of true or false.  So, I added this condition here.

Reply:
It's because it is the wrong type of enum. The other type of enum will remove the implicit coercions that are causing the problem here.